infra/scoped/run_terraform.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+TERRAFORM_WORKSPACE=$(terraform workspace show)
+
+AWS_CLI_PROFILE="identity-terraform"
```

Comment:
Instead of relying on a user to set the correct profile as in https://github.com/wellcomecollection/platform-infrastructure/blob/main/critical/run_terraform.sh, we create a custom profile and assign it a known role.